# Coding Exercises (Part 1)

Now, you will have the opportunity to practice what you have learned. <br>
__Follow the instructions__ and insert your code! 

The correct answer is provided below your coding cell. There you can check whether your code is correct.

If you need some help or if you want to check your code, you can also have a look at the __solutions__.

### Have Fun!

--------------------------------------------------------------------------------------------------------------

## Exercise 1: Using Python as a Calculator for TVM Problems

Use __variables__ where appropriate! __Input__ values and your __final results__ shall be stored in variables. 

1. You save 2,000 USD for 7 years at an interst rate of 4% p.a.. Calculate the value of your account in 7 Years (__Future Value__)!

In [ ]:
# Store Input values in variables


In [20]:
pv = 2000 # present value
n = 7 # number of periods
r = 0.04 # interest rate

In [21]:
fv = PV * (1 + r)**n

In [22]:
fv

2631.8635584716803

In [ ]:
# Calculate and store the future value in a variable


In [23]:
# print the future value
print(PV * (1 + r)**n)

2631.8635584716803


(result: 2631.86 USD)

2. You consider to purchase a car in 4 years for 20,000 USD. How many USD do you have to save today (__Present Value__) in your savings account to have 20,000 USD in 4 years? Assume an interst rate of 3.5% p.a. 

In [38]:
# Store Input values in variables

fv = 2000
n = 4
r = .035

In [39]:
# Calculate and store the present value in a variable
pv = fv /(1 + r)**n

In [40]:
pv

1742.8844553971448

In [41]:
# print the present value
print (FV/((1 + r)**n))

1742.8844553971448


(result: 17,428.84 USD)

3. Bank B offers you to save 10,000 USD today, getting back 15,000 USD in 10 Years. Calculate the __per annum compound interest rate__. Bank A offers you an interest rate of 4.5% p.a.. Which offer should you accept?

In [35]:
pv = 10000
fv = 15000
n = 10

In [36]:
r = (fv/pv)**(1/n) - 1

In [37]:
r

0.04137974399241062

(result: 4.14%)

4. One year ago you invested 976 USD in the ABC stock. Recently you received a 40 USD Dividend Payment and the current value of your ABC stock investment is 1,053 USD. Calculate the __Price Return__! 

In [43]:
p0 = 976
p1 = 1053
div = 40

In [44]:
p_ret = p1/p0 -1

In [45]:
p_ret

0.07889344262295084

(result: 7.89%)

5. Calculate __Dividend Yield__ and __Total Return__ for your ABC stock investment!

In [46]:
div_y = div/p0

In [47]:
div_y

0.040983606557377046

In [48]:
total_ret = p_ret + div_y

In [49]:
total_ret

0.11987704918032788

(results: 4.10% and 11.99%)

6. Print Price Return, Dividend Yield and Total Return in one single line, __separated by commas__!

In [52]:
print (p_ret, div_y, total_ret, sep = ", ")

0.07889344262295084, 0.040983606557377046, 0.11987704918032788


# Well Done!

---------------------------------------------------------------------------------------------------------------------

# Solutions (Stop here if you want to code on your own!)

# ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

1. You save 2,000 USD for 7 years at an interst rate of 4% p.a.. Calculate the value of your account in 7 Years (__Future Value__)!

In [ ]:
# Store Input values in variables
pv = 2000
n = 7
r = 0.04

In [ ]:
# Calculate and store the future value in a variable
fv = pv * (1 + r)**n

In [ ]:
# print the future value
fv

(result: 2631.86 USD)

2. You consider to purchase a car in 4 years for 20,000 USD. How many USD do you have to save today (__Present Value__) in your savings account to have 20,000 USD in 4 years? Assume an interst rate of 3.5% p.a. 

In [ ]:
# Store Input values in variables
fv = 20000
n = 4
r = 0.035

In [ ]:
# Calculate and store the present value in a variable
pv = fv / (1 + r)**n

In [ ]:
# print the present value
pv

(result: 17,428.84 USD)

3. Bank B offers you to save 10,000 USD today, getting back 15,000 USD in 10 Years. Calculate the __per annum compound interest rate__. Bank A offers you an interest rate of 4.5% p.a.. Which offer should you accept?

In [ ]:
# Store Input values in variables
pv = 10000
fv = 15000
n = 10

In [ ]:
r = (fv / pv)**(1/n) - 1

In [ ]:
r

(result: 4.14%)

4. One year ago you invested 976 USD in the ABC stock. Recently you received a 40 USD Dividend Payment and the current value of your ABC stock investment is 1,053 USD. Calculate the __Price Return__! 

In [ ]:
p0 = 976
p1 = 1053
div = 40

In [ ]:
p_ret = p1 / p0 - 1

In [ ]:
p_ret

(result: 7.89%)

5. Calculate __Dividend Yield__ and __Total Return__ for your ABC stock investment!

In [ ]:
div_y = div / p0

In [ ]:
div_y

In [ ]:
total_ret = p_ret + div_y

In [ ]:
total_ret

(results: 4.10% and 11.99%)

6. Print Price Return, Dividend Yield and Total Return in one single line, __separated by commas__!

In [ ]:
print(p_ret, div_y, total_ret, sep = ", ")